### PR TITLE
bpo-29710: Clarify documentation for Bitwise binary operation

### PR DESCRIPTION
**Pull request by CuriousLearner**
Resolves http://bugs.python.org/issue29710 

clarify documentation for bitwise binary operators.

https://bugs.python.org/issue29710

### CODE REVIEW

**Comment by CuriousLearner** [comment]
Hey @terryjreedy , @birkenfeld & @ncoghlan !

This patch is pending for a while. Can you please take some time to review this? :)

File: Doc/library/stdtypes.rst [hunk 1]
```diff
@@ -441,6 +440,11 @@ Notes:
    A right shift by *n* bits is equivalent to division by ``pow(2, n)`` without
    overflow check.
 
+(4)
```
Comment (vadmium) [inline Doc/library/stdtypes.rst:441]: Forgot to add references to the new footnote?

Comment (CuriousLearner) [reply Doc/library/stdtypes.rst:441]: @vadmium How should I reference it? Is this what is required?

```
index 5fa3ca10fd..40129f0037 100644
--- a/Doc/library/stdtypes.rst
+++ b/Doc/library/stdtypes.rst
@@ -400,7 +400,7 @@ Bitwise Operations on Integer Types
    operator: >>

 Bitwise operations only make sense for integers.  Negative numbers are treated
-as their 2's complement value.
+as their 2's complement value (4).
```

Comment (vadmium) [reply Doc/library/stdtypes.rst:441]: If you are trying to follow Nick’s suggestion (move the note about negative numbers to a numbered footnote in the table), I think he was suggesting adding (4) to the table that the other three footnotes relate to, specifically to each of the first three rows for x | y, x ^ y, and x & y. These are bitwise binary operations and they define _x_ and _y_.

However I am not enthusiastic about the new text; see my comments in the bug thread.

Comment (CuriousLearner) [reply Doc/library/stdtypes.rst:441]: So, what do you think about Nick's suggestion, where he included Mark's suggestion as well: 

``` Each bitwise operation has the same result as though carried out in two's complement using a bit-width that's large enough to represent the inputs. ("Large enough" for this purpose is ``1 + max(x.bit_length(), y
.bit_length()``, with the extra bit being needed to handle sign extension appropriately)```

**Comment by CuriousLearner** [comment]
@vadmium @brettcannon Is there anything I can do here to get this merged?

**Review by ncoghlan — CHANGES_REQUESTED** [review]
See the bpo issue for further proposed wording updates.

**Comment by CuriousLearner** [comment]
@ncoghlan  I've made the changes as suggested by you. Can you please have a look now? :)

**Comment by brettcannon** [comment]
@CuriousLearner do note that the bot requests you say "I have made the requested changes; please review again" in order to signal to @ncoghlan and other core devs that you are ready to have this PR looked at again.

**Comment by CuriousLearner** [comment]
I have made the requested changes; please review again

**Review by ncoghlan — APPROVED** [review]
Thanks, and thanks for your patience with the long discussion of the precise wording!

**Pull request merged**